resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -438,7 +443,7 @@ private[spark] class KubernetesClusterSchedulerBackend(
             .addToRequests("memory", executorMemoryQuantity)
             .addToLimits("memory", executorMemoryLimitQuantity)
             .addToRequests("cpu", executorCpuQuantity)
-            .addToLimits("cpu", executorCpuQuantity)
+            .addToLimits("cpu", executorCpuLimitQuantity)
```

Comment:
@sandflee can you please change this PR so that if `spark.kubernetes.executor.limit.cores` is not set then there is no `  .addToLimits("cpu", executorCpuLimitQuantity)` call here?

From discussion on the linked issue I think we want the default to be no cpu limit, and only add a limit if the user specifies one